File: RxAndroid/app/src/main/java/com/helen/rxandroid/MainActivity.java
package com.helen.rxandroid;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.helen.rxandroid.presenter.MainPresenter;
import com.helen.rxandroid.view.IMainView;

public class MainActivity extends AppCompatActivity implements IMainView{

    private TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = findViewById(R.id.text);
        MainPresenter mainPresenter = new MainPresenter(this);
        mainPresenter.getData();

    }



    @Override
    public void getData(String data) {
        if (textView!=null){
            textView.setText(data);
        }
    }
}

File: MyTest/app/src/main/java/com/helen/mytest/MainActivity.kt
package com.helen.mytest

import android.widget.TextView
import com.helen.baselib.base.BaseActivity
import com.helen.baselib.commonwidget.LoadingDialog
import com.helen.mytest.test.TestContract
import com.helen.mytest.test.TestModel
import com.helen.mytest.test.TestPresenter

class MainActivity : BaseActivity<TestPresenter,TestModel>(), TestContract.View {
    var text:TextView? = null;
    override fun showView(test: String) {
        if (text!=null) text!!.setText(test)
    }

    override fun showLoading(title: String?) {
        LoadingDialog.showDialogForLoading(this@MainActivity,title,true)
    }

    override fun stopLoading() {
        LoadingDialog.cancelDialogForLoading();
    }

    override fun showErrorTip(msg: String?) {
        showErrorTip(msg)
    }

    override fun getLayoutResId(): Int {
      return  R.layout.activity_main
    }

    override fun initPresenter() {
        mPresenter.setVM(mModel,this@MainActivity)
    }

    override fun initView() {
        text = findViewById<TextView>(R.id.text);
        mPresenter.setdata()
    }


}
